src/modules/ekf2/EKF/aid_sources/magnetometer/mag_control.cpp

```diff
@@ -69,12 +69,22 @@ void Ekf::controlMagFusion(const imuSample &imu_sample)
 
 			_state.mag_B.zero();
 			resetMagBiasCov();
+			_state.mag_I.zero();
+			resetMagEarthCov();
 
 			stopMagFusion();
 
 			_mag_lpf.reset(mag_sample.mag);
 			_mag_counter = 1;
 
+			if (!_control_status.flags.in_air
+			    && ((_params.mag_fusion_type == MagFuseType::INIT)
+				|| (_params.mag_fusion_type == MagFuseType::AUTO)
+				|| (_params.mag_fusion_type == MagFuseType::HEADING))) {
+				// Clear yaw alignment to force a reset when mag calibration changed
+				_control_status.flags.yaw_align = false;
```

Comment:
Let clearing yaw_align be handled by stopMagFusion()? https://github.com/PX4/PX4-Autopilot/blob/519b1f067c787da0519405ae536c90fad32bcaea/src/modules/ekf2/EKF/aid_sources/magnetometer/mag_control.cpp#L334

Reply:
Same a for the other one, I think this is quite specific to a change of mag calibration parameters

Reply:
Isn't this logic slightly wrong though resetting yaw_align based on the param rather than `_control_status.flags.mag_hdg`/`_control_status.flags.mag_3D`? https://github.com/PX4/PX4-Autopilot/blob/519b1f067c787da0519405ae536c90fad32bcaea/src/modules/ekf2/EKF/aid_sources/magnetometer/mag_control.cpp#L330C43-L336

Reply:
Oh, I didn't realize we had so many things in `stopMagFusion`. Let me see if all of this really makes sense.

Reply:
So, the reason why I used the params an not the control flags is because the yaw can be aligned when starting `mag` fusion but `mag_hdg` or `mag_3D` are not starting (e.g.: mag type "INIT").
Maybe a better solution would be to check for 
`_control_status.flags.mag && !(_params.mag_fusion_type == MagFuseType::NONE)`

Reply:
In fact we don't even arrive here when we are in mag type "None", so we don't need to care about this.
I simplified the condition. I think that one should really stay out of the `stopMagFusion` code as it is directly related to a reset and not to the general stop/start logic.

Reply:
I was also thinking about `gnss_yaw`, but this is a pretty minor edge case either way.